src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -277,3 +280,92 @@ def _check_authorization(self, username):
         if not AuthManager.authorize(username, sec_scope, sec_perms):
             raise cherrypy.HTTPError(403, "You don't have permissions to "
                                           "access that resource")
+
+class Oauth2JWTManager:
+    _token = None
+    _token_payload = None
+    _user = None
```

Comment:
I'd bet you to test this with multiple Dashboard users logged in... Sooner or later 2 threads will concurrently overwrite this and you'll get credentials from 1 user accessed by the other one.

That's why the JWTManager had (which was a clumsy thing, since `cherrypy.request` is already a thread-local object):
```py
    LOCAL_USER = threading.local()
```

Reply:
Using `cherrypy.request` now